stdlib/Artifacts/src/Artifacts.jl

```diff
@@ -176,6 +161,23 @@ function load_overrides(;force::Bool = false)::Dict{Symbol, Any}
     return overrides
 end
 
+# These could be inner functions in load_overrides, except the `parse_mapping(v, name) for (k,v) in mapping` closure
+# below triggers #15276, specifically the #23618 flavor
+function parse_mapping(mapping::String, name::String)
+    if !isabspath(mapping) && !isempty(mapping)
+        mapping = tryparse(Base.SHA1, mapping)
+        if mapping === nothing
+            @error("Invalid override in '$(override_file)': entry '$(name)' must map to an absolute path or SHA1 hash!")
```

Comment:
`override_file` exists only in the scope of the loop in which this function lived originally I think?

Reply:
Hmm, that is a bit of a problem :astonished: